jdbc/src/main/scala/zio/sql/jdbc.scala

```diff
@@ -184,8 +185,21 @@ trait Jdbc extends zio.sql.Sql {
             column.fold(resultSet.getTimestamp(_), resultSet.getTimestamp(_)).toLocalDateTime().toLocalTime()
           )
         case TLong               => tryDecode[Long](column.fold(resultSet.getLong(_), resultSet.getLong(_)))
-        case TOffsetDateTime     => ???
-        case TOffsetTime         => ???
+        case TOffsetDateTime     =>
+          tryDecode[OffsetDateTime](
+            column
+              .fold(resultSet.getTimestamp(_), resultSet.getTimestamp(_))
+              .toLocalDateTime()
+              .atOffset(ZoneOffset.of(ZoneId.systemDefault().getId))
+          )
+        case TOffsetTime         =>
+          tryDecode[OffsetTime](
+            column
+              .fold(resultSet.getTimestamp(_), resultSet.getTimestamp(_))
```

Comment:
I believe we need `resultSet.getTime` for this - I'm not sure what calling `getTimestamp` on a time column would do.